clients/registry-amqp/src/test/java/org/eclipse/hono/client/registry/amqp/ProtonBasedCredentialsClientTest.java

```diff
@@ -461,12 +413,14 @@ public void testAllDevicesOfTenantDeletedNotificationRemovesValueFromCache(final
                 .onComplete(ctx.succeeding(ok -> ctx.verify(() -> {
 
                     // WHEN receiving a notification that all devices of a tenant have been deleted
-                    notificationHandlerCaptor.getValue()
-                            .handle(new AllDevicesOfTenantDeletedNotification(tenantId, Instant.now()));
+                    NotificationEventBusSupport.sendNotification(vertx,
+                            new AllDevicesOfTenantDeletedNotification(tenantId, Instant.now()));
 
                     // THEN the cache is invalidated for all credentials of the tenant and not for other tenants
-                    ctx.verify(() -> verify(cache).invalidateAll(expectedCacheRemovals));
-                    ctx.completeNow();
+                    vertx.runOnContext(v -> vertx.executeBlocking(p -> {
```

Comment:
I guess this needs to be done in order to make sure that the event has been already been sent and processed? IMHO this test should be implemented using a mocked Vertx instance where you have real control over what happens when ...

Reply:
I originally wanted to keep using a mocked Vertx instance here, but found it to be rather complicated. With the removal of the `NotificationEventBusContainer` class it's a bit easier.
I've amended the commit accordingly.